backend/LeetWars.Builder/Services/SolutionRunnerService.cs

```diff
@@ -1,29 +1,142 @@
 ﻿using Docker.DotNet;
 using Docker.DotNet.Models;
 using LeetWars.Builder.Interfaces;
+using LeetWars.Builder.Models;
+using LeetWars.Builder.RunnerDefaults;
+using LeetWars.Builder.RunnerDefaults.CSharp;
+using LeetWars.Builder.RunnerDefaults.JS;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
 
 namespace LeetWars.Builder.Services
 {
     public class SolutionRunnerService : ISolutionRunnerService, IDisposable
     {
         private readonly DockerClient _client = new DockerClientConfiguration().CreateClient();
 
-        public async Task RunCodeInContainerAsync(string imageName, string containerName)
+        private readonly IXmlTestResultParserService _parserService;
+
+        private readonly ITarManagementService _tarManagementService;
+
+        private readonly JsonSerializerSettings _serializerSettings = new() { ContractResolver = new CamelCasePropertyNamesContractResolver() };
+
+        private readonly string _localVolumeName = "LocalVolume";
+
+
+    public SolutionRunnerService(IXmlTestResultParserService parserService, ITarManagementService tarManagementService)
         {
-            var containerParams = new CreateContainerParameters
+            _parserService = parserService;
+
+            _tarManagementService = tarManagementService;
+        }
+
+        public async Task<string> RunSolutionTestsAsync(string processName, string language, string code, string tests, string preloaded)
+        {
+            return language switch
             {
-                Image = imageName,
-                Name = containerName
+                "csharp" => JsonConvert.SerializeObject(await RunCSharpSolutionTestsAsync(processName, code, tests, preloaded), _serializerSettings),
```

Comment:
It would be better to create shared const and use them everywhere

Reply:
Separated to a file with consts